langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Sort.java

```diff
@@ -58,15 +68,20 @@ public static BArray sort(BArray arr, Object direction, Object func) {
                 sortArr[i][0] = sortArr[i][1] = arr.get(i);
             }
         }
-
         mergesort(sortArr, sortArrClone, 0, sortArr.length - 1, direction.toString());
-
-        BArray sortedArray = ValueCreator.createArrayValue(TypeCreator.createArrayType(arr.getElementType()));
-
+        if (arrType.getTag() == TypeTags.TUPLE_TAG) {
+            Set<Type> typeList = new HashSet<>(((BTupleType) arrType).getTupleTypes());
```

Comment:
define a new variable for `(BTupleType) arrType`, and us that variable in other places

Reply:
Fixed with 5f118b816390f07e4e19f2f677907234a7348a78